applications/FluidDynamicsApplication/python_scripts/navier_stokes_compressible_explicit_solver.py

```diff
@@ -150,18 +150,24 @@ def _create_solution_strategy(self):
             err_msg = err_msg + " - {}\n".format(key)
         raise RuntimeError(err_msg)
 
+    @classmethod
+    def _OverrideBoolParameterWithWarning(cls, parent, child, value):
+        if parent.Has(child) and parent[child].GetBool() != value:
+            KratosMultiphysics.Logger.PrintWarning("", "User-specifed {} will be overriden with {}".format(child, value))
```

Comment:
```suggestion
        if parent.Has(child):
            if parent[child].GetBool() != value:
                KratosMultiphysics.Logger.PrintWarning("", "User-specifed {} will be overriden with {}".format(child, value))
```

Reply:
Why? I think the original version looks better

Reply:
Wouldn't it crash if the `Has` is false when trying to access `parent[child]`? (here I'm assuming that all the loop clauses are evaluated, not 100% sure about how it works internally).

Reply:
AFAIK, `and` compound statements does not execute the trailing ones if one of the leading ones are false.

```
class A:
   def Is(self, t):
      print("test")
      return t
a = A()
b = A()
if a.Is(False) and b.Is(False):
   print("D")
```

Reply:
nope, "and" and "or" operators follow short-circuit evualtion. In the and, if first condition is false it does not access the second one. Similarly, in or, if first condition is true, second is not evaluated.

Reply:
https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not

Reply:
Python has short-circuiting just like C and C++.

With the following statement:
```Python
boolean = condition1() and condition2_with_side_effects()
```

If `condition1()` evaluates to true, `condition2` is never evaluated and the side effects never take place.

See the [reference](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) regarding the `and` operator:
> This is a short-circuit operator, so it only evaluates the second argument if the first one is true.

Reply:
pole

Reply:
Very good. Thanks. I wasn't aware.